### PR TITLE
Tno 2657 prevent sending empty report

### PR DESCRIPTION
**Pull request by kkwangsir**
Check empty Front Page Images report before send, and make toast messages.

### CODE REVIEW

**Review by Fosol — APPROVED** [review]

File: app/editor/src/features/content/papers/PaperToolbar.tsx
Comment (Fosol) [inline app/editor/src/features/content/papers/PaperToolbar.tsx:1]: I would have though the preview will have some HTML still, but no content?

Comment (kkwangsir) [reply app/editor/src/features/content/papers/PaperToolbar.tsx:1]: image

yes, I copied part of  preview details.

`{
    "subject": "\n",
    "body": "\n",
`

Comment (Fosol) [reply app/editor/src/features/content/papers/PaperToolbar.tsx:1]: This doesn't look correct to me.  Make certain your report template is valid.

Comment (kkwangsir) [reply app/editor/src/features/content/papers/PaperToolbar.tsx:1]: image
Yes, You are right. I found I only have three. Should I create Front Pages report manually or import from somewhere?

Comment (Fosol) [reply app/editor/src/features/content/papers/PaperToolbar.tsx:1]: You can probably copy it from the TEST environment manually.  Not sure why that report doesn't exist in our local setup.  You may want to fix that issue to.

Comment (kkwangsir) [reply app/editor/src/features/content/papers/PaperToolbar.tsx:1]: Thanks. I copied it into my machine. and change the setting default front page report id,

Comment (Fosol) [reply app/editor/src/features/content/papers/PaperToolbar.tsx:1]: The issue I see here is that the report that is generated will still have HTML, it simply will not have any content.  Which will make the below code fail.

**Comment by kkwangsir** [comment]
image
image
I spend some time to reviewing the backend code and and made modifications to parse the report preview data, which includes content results. I make a few tests using both empty and populated reports, and the system appears to be functioning correctly, even when handling reports that contain HTML but lack actual content

File: app/editor/src/features/content/papers/PaperToolbar.tsx [hunk 1]
```diff
@@ -41,18 +42,76 @@ export interface IPaperToolbarProps {
...
+  const [isLoading, setIsLoading] = React.useState(false);
+
   React.useEffect(() => {
     setMediaTypeOptions([new OptionItem<number>('Any', 0), ...mediaTypeOptions]);
   }, [mediaTypeOptions]);
 
+  async function checkReportContent(reportID: number) {
+    try {
+      setIsLoading(true);
+      const preview = await previewReport(reportID);
```
Comment (Fosol) [inline app/editor/src/features/content/papers/PaperToolbar.tsx:62]: Can you check if the report has the message in it "There is no content in this report."?

Comment (kkwangsir) [reply app/editor/src/features/content/papers/PaperToolbar.tsx:62]: OK. No problem.

**Review by Fosol — APPROVED** [review]

**Pull request merged**